hs_core/views/resource_rest_api.py

```diff
@@ -299,7 +299,7 @@ def get_queryset(self):
         filter_parms['public'] = not self.request.user.is_authenticated()
         filtered_res_list = []
 
-        for r in hydroshare.get_resource_list(**filter_parms):
+        for r in self.paginate_queryset(hydroshare.get_resource_list(**filter_parms)):
```

Comment:
This is the big win in optimization.  Before this, all resources returned in the queryset were being serialized, and then pagination was applied.  This change pages the results before serialization.